### PR TITLE
Realm.io final content rescue updates

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
## Pull Request Info
- Based on Christian's feedback to other realm.io final content rescue tickets.

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/christian_updates/sdk/android/fundamentals/live-queries/#managed-objects (managed objects and unmanaged objects sections)
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/christian_updates/sdk/android/fundamentals/realms/#the-realm-lifecycle (copy updates around the middle of "The Realm Lifecycle" section)

### Review Guidelines
[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/android/fundamentals/realms.txt [hunk 1]
```diff
@@ -111,6 +111,34 @@ maximize the performance of your application, you should minimize the
...
+  same process, opening the {+realm+} is less expensive, but still
+  nontrivial.
+
+- If the {+realm+} is already open on the same thread within the same
+  process, opening the {+realm+} requires minimal additional resources.
+
+When you open a {+realm+} for the first time, {+client-database+}
+performs the memory-mapping and schema validation required to read and
+write data to the {+realm+}. Additional instances of that
+{+realm+} on the same thread uses the same underlying resources.
```
Comment (biniona-mongodb) [inline source/sdk/android/fundamentals/realms.txt:132]:
```suggestion
{+realm+} on the same thread use the same underlying resources.
```

Should "uses" be "use"?   "uses" sounds incorrect to my ear, but could be wrong

File: source/sdk/android/fundamentals/realms.txt [hunk 2]
```diff
@@ -111,6 +111,34 @@ maximize the performance of your application, you should minimize the
...
 
+However, opening a {+realm+} is not always consistently expensive.
+The amount of resources consumed when opening a {+realm+} instance is
+determined by whether that {+realm+} is already open within the
+same process and thread:
+
+- If the {+realm+} is not open within the same process, opening the
+  {+realm+} is expensive.
+
+- If the {+realm+} is already open on a different thread within the
```
Comment (biniona-mongodb) [inline source/sdk/android/fundamentals/realms.txt:122]:
```suggestion
- If the {+realm+} is open on a different thread within the
```

Nit:

I am not sure what "already" adds to the sentence. When I took the word out, it made the sentence easier to understand for me (removes dimension of time). If there is some meaning it adds that is necessary to the concept however, totally understand.

Comment (nathan-contino-mongo) [reply source/sdk/android/fundamentals/realms.txt:122]: Probably not really required, but for some reason it makes more sense to me written this way. I'll see if I can reword this to eliminate this entirely.

File: source/sdk/android/fundamentals/live-queries.txt [hunk 3]
```diff
@@ -132,6 +132,29 @@ between the objects in {+client-database+} and the results of your query in
...
+.. _android-unmanaged-objects:
+
+Unmanaged objects
+-----------------
+
+**Unmanaged objects** are instances of {+service-short+} objects that are
+not live. You can get an unmanaged object by manually constructing a
+{+service-short+} object yourself, or by calling
+:java-sdk:`Realm.copyFromRealm() <io/realm/Realm.html#copyFromRealm-E->`.
+Unmanaged objects are thread-safe and as a result can be passed between
```
Comment (biniona-mongodb) [inline source/sdk/android/fundamentals/live-queries.txt:155]:
```suggestion
Unmanaged objects are thread-safe and can be passed between
```
Nit:

Not sure what "and as a result" adds, if you think it is helpful though totally understand

Comment (nathan-contino-mongo) [reply source/sdk/android/fundamentals/live-queries.txt:155]: "can be passed between threads" is really just a more intuitive way of saying "thread-safe" and was trying to connect those two thoughts. But in hindsight I can just say they're thread-safe and let the reader figure it out from there.

File: source/sdk/android/fundamentals/realms.txt (hunk 2 above)
Comment (nathan-contino-mongo) [reply source/sdk/android/fundamentals/realms.txt:121]: Ended up rewording the introduction for this list in a way that ties "already" into the narrative a bit more.

File: source/sdk/android/advanced-guides/threading.txt [hunk 4]
```diff
@@ -117,12 +117,12 @@ several options depending on your use case:
...
-and then open a new {+realm+} instance in the separate thread to query
-for that identifier.
+:ref:`Managed <android-managed-objects>` ``RealmObject`` instances are
+not thread-safe or ``Parcelable``, so you cannot pass them between
+activities or threads via an ``Intent``. Instead, you can pass an object
+identifier, like a :ref:`primary key <android-primary-key-fundamentals>`,
+in the ``Intent`` extras bundle, and then open a new {+realm+} instance
+in the separate thread to query for that identifier.
 
 .. seealso::
```
Comment (MongoCaleb) [inline source/sdk/android/advanced-guides/threading.txt:127]: A code snippet here that shows adding the ID to the ``extras`` bundle might be useful. (?)

File: source/sdk/android/fundamentals/live-queries.txt [hunk 5]
```diff
@@ -132,6 +132,28 @@ between the objects in {+client-database+} and the results of your query in
...
+.. _android-managed-objects:
+
+Managed Objects
+---------------
+
+**Managed objects** are live {+service-short+} objects that update
+based on changes to underlying data in {+client-database+}. Managed
+objects can only come from an open {+realm+}, and receive updates
+as long as that {+realm+} remains open. Managed objects cannot be passed
+between threads.
```
Comment (MongoCaleb) [inline source/sdk/android/fundamentals/live-queries.txt:144]: This feels like an abrupt transition. Should it be in a note, perhaps? Or a new paragraph with another sentence or two?

File: source/sdk/android/fundamentals/live-queries.txt [hunk 6]
```diff
@@ -132,6 +132,28 @@ between the objects in {+client-database+} and the results of your query in
...
+.. _android-unmanaged-objects:
+
+Unmanaged objects
+-----------------
+
+**Unmanaged objects** are instances of {+service-short+} objects that are
+not live. You can get an unmanaged object by manually constructing a
+{+service-short+} object yourself, or by calling
+:java-sdk:`Realm.copyFromRealm() <io/realm/Realm.html#copyFromRealm-E->`.
+Unmanaged objects are thread-safe.
```
Comment (MongoCaleb) [inline source/sdk/android/fundamentals/live-queries.txt:155]: > Unmanaged objects are thread-safe, and so can be passed between threads

Maybe?

File: source/sdk/android/fundamentals/realms.txt (hunk 1 above)
Comment (MongoCaleb) [reply source/sdk/android/fundamentals/realms.txt:131]: "use" not "uses"

**Review by MongoCaleb — APPROVED** [review]
a few suggestions you can feel free to ignore.

File: source/sdk/android/advanced-guides/threading.txt (hunk 4 above)
Comment (nathan-contino-mongo) [reply source/sdk/android/advanced-guides/threading.txt:127]: The original realm.io text does include this, but I wonder if it's truly necessary since bundles/extras are such basic Android patterns (it would sort of be like showing someone how to call a function in their java code in a non-android codebase)

File: source/sdk/android/fundamentals/live-queries.txt (hunk 5 above)
Comment (nathan-contino-mongo) [reply source/sdk/android/fundamentals/live-queries.txt:144]: I'll bump both threading mentions into notes. Thanks!

File: source/sdk/android/fundamentals/live-queries.txt (hunk 6 above)
Comment (nathan-contino-mongo) [reply source/sdk/android/fundamentals/live-queries.txt:155]: Ha, I actually removed pretty much that exact thing before but I think I'm going to restore it now.
